core/src/smartcontracts/isi/asset.rs

```diff
@@ -361,14 +361,57 @@ pub mod query {
     impl ValidQuery for FindAllAssets {
         #[metrics(+"find_all_assets")]
         fn execute(&self, wsv: &WorldStateView) -> Result<Self::Output, Error> {
+            let sort_by_key = match self.sort_by_key.as_ref() {
```

Comment:
Move sorting to where the `pagination` and `filter` are applied in the query endpoint in `torii`. 

Sorting should be applied before all other operations. 

That way, you don't need to add sorting to all queries and all queries are sorted the same way.